lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -4368,9 +4372,39 @@ LogicalResult FIRRTLLowering::lowerVerificationStatement(
     Value opEnable, StringAttr opMessageAttr, ValueRange opOperands,
     StringAttr opNameAttr, bool isConcurrent, EventControl opEventControl) {
   StringRef opName = op->getName().stripDialect();
+
+  // The attribute holding the compile guards
+  ArrayRef<Attribute> guards{};
+  if (auto guardsAttr = op->template getAttrOfType<ArrayAttr>("guards"))
+    guards = guardsAttr.getValue();
+
+  // Get the casted version of guards to concrete StringRef type
+  // Need to first use a vector container to do the transform and then wrap it
+  // around an ArrayRef.
+  std::vector<StringRef> guardsStringRefVector(guards.size());
```

Comment:
LLVM coding style tends to use in-tree data structures as opposed to `std` ones (with a few exceptions). Almost anyplace you want to use `std::vector<A>`, you can use `SmallVector<A>`. The latter has some benefits in that it will try to be stack allocated until it can't. The [LLVM Programmer's Manual](https://llvm.org/docs/ProgrammersManual.html) gets into a lot of discussion on this.

Reply:
Going by the other comments - for this PR at least I will no longer  need `std::vector` - but will check the docs for future reference!